core/src/main/kotlin/com/karumi/kotlinsnapshot/core/SerializationModule.kt

```diff
@@ -0,0 +1,65 @@
+package com.karumi.kotlinsnapshot.core
+
+import kotlin.reflect.KProperty1
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.memberProperties
+
+interface SerializationModule<in A> {
+
+    fun serialize(value: A?): String
+}
+
+class KotlinSerialization : SerializationModule<Any> {
+
+    override fun serialize(value: Any?): String = when {
+        value == null -> "null"
```

Comment:
I have an open question for this scenario, should we include the variable's type name? Should a test fail if we are expecting something like a `User` and receive an `Admin` instead, even if they are null? How annoying would it be if we fail but forget about updating it when renaming classes? 🤔 

I'm going to say we should be adding it, we are already doing it in some of the types below, e.g. We are using different representations for arrays and maps.

Reply:
Yes, that's would be cool, how could look like? `User(null)` or something like that?

Reply:
That sounds about right! 👍

Reply:
We discussed about it in person and there is an important missing piece I didn't thought about: we don't have access to the runtime type of a null value so we will have to think about it in-depth if we want to make it possible in the future. Let's leave it for a different PR.